### PR TITLE
Upgrade to  node 20.12.2

### PR DESCRIPTION
**Pull request by Timodie**
### Description
Modeled after https://github.com/silvermine/videojs-airplay/pull/54
This PR upgrades this project to node 20.12.2 but does not update the grunt-contrib-uglify version

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -5,34 +5,54 @@ on: [ push, pull_request ]
...
-         - run: npm i -g npm@8.5.5
+         -
+            name: PUT NVM version in output
+            id: makeNodeVersionOutput
+            run: echo "nvmrc=$(cat .nvmrc)" >> "$GITHUB_OUTPUT"
+
          - run: npm ci
+         - run: npm run check-node-version
          - run: npm run standards
+         - run: npm run build
```
Comment (kmuncie) [inline .github/workflows/ci.yml:27]: Is this missing the "Check for uncommitted changes" on purpose?

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -5,34 +5,54 @@ on: [ push, pull_request ]
...
-            uses: actions/setup-node@v3
+            uses: actions/setup-node@v4
             with:
                node-version-file: '.nvmrc'
-         - run: npm i -g npm@8.5.5
+         -
+            name: PUT NVM version in output
+            id: makeNodeVersionOutput
+            run: echo "nvmrc=$(cat .nvmrc)" >> "$GITHUB_OUTPUT"
+
```
Comment (kmuncie) [inline .github/workflows/ci.yml:23]: is this new line intended?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -5,34 +5,54 @@ on: [ push, pull_request ]
...
          - run: npm run standards
+         - run: npm run build
    test:
       needs: [ build ]
       runs-on: ubuntu-latest
       strategy:
          fail-fast: false
          matrix:
-            node-version: [ 14, 16, 'lts/*', 'latest' ]
+            node-version: [ 16,'${{ needs.build.outputs.nvmrc }}', 'lts/*', 'latest' ]
```
Comment (kmuncie) [inline .github/workflows/ci.yml:34]: Whitespace issue after the first comma

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -5,34 +5,54 @@ on: [ push, pull_request ]
...
+         - uses: actions/checkout@v4
          -
             name: Use Node.js ${{ matrix.node-version }}
-            uses: actions/setup-node@v3
+            uses: actions/setup-node@v4
             with:
                node-version: ${{ matrix.node-version }}
-         - run: npm i -g npm@8.5.5
          - run: npm ci # Reinstall the dependencies to ensure they install with the current version of node
          - run: npm test
```
Comment (kmuncie) [inline .github/workflows/ci.yml:43]: Should we be running the following before `npm test`?

```
         - run: npm run standards
         - run: npm run build # Ensure building is possible with this version of node
```

File: .nvmrc [hunk 5]
```diff
@@ -1 +1 @@
-16.15.0
+20.12.2
```
Comment (kmuncie) [inline .nvmrc:1]: I believe we are standardizing on including the `v` here

File: Gruntfile.js [hunk 6]
```diff
@@ -91,9 +91,9 @@ module.exports = function(grunt) {
                banner: '/*! <%= pkg.name %> <%= grunt.template.today("yyyy-mm-dd") %> <%= versionInfo %> */\n',
                sourceMap: true,
```
Comment (kmuncie) [inline Gruntfile.js:92]: I believe we also usually use DEBUG on this line and the following

File: package.json [hunk 7]
```diff
@@ -4,8 +4,11 @@
   "description": "video.js plugin for selecting a video quality or resolution",
   "main": "src/js/index.js",
   "scripts": {
-    "prepare": "grunt build",
-    "test": "check-node-version --npm 8.5.5 && nyc mocha -- 'tests/**/*.test.js'",
+    "prepare": "npm run build",
```
Comment (kmuncie) [inline package.json:7]: The name `prepare` seems out of norm compared to our other projects. Should this be changed to `prepublish`?

File: package.json [hunk 8]
```diff
@@ -4,8 +4,11 @@
...
+    "check-node-version": "check-node-version --node $(cat .nvmrc) --npm 10.5.0 --print",
+    "test": "nyc mocha -- -R spec 'tests/**/*.test.js'",
```
Comment (kmuncie) [inline package.json:9]: I don't see the `-R` "reporter" option configured in many other projects, is that intended to be added here where it wasnt present before?

**Review by kmuncie — COMMENTED** [review]
Just a first pass review, @joshuacurtiss can take it from here

**Comment by joshuacurtiss** [comment]
Please reference #112.

**Comment by onebytegone** [comment]
Superseded by https://github.com/silvermine/videojs-quality-selector/pull/112

**Pull request closed without merging**